src/gobject.cc

```diff
@@ -453,45 +452,29 @@ static MaybeLocal<FunctionTemplate> GetClassTemplate(GIBaseInfo *gi_info, GType
         return tpl;
     }
 
-    if (gi_info == NULL)
-        gi_info = g_irepository_find_by_gtype(NULL, gtype);
-
-    assert_printf (gi_info != NULL, "Missing GIR info for: %s\n", g_type_name (gtype));
-
-    auto maybeTpl = NewClassTemplate(gi_info, gtype);
+    auto maybeTpl = NewClassTemplate(gtype);
     if (maybeTpl.IsEmpty())
         return MaybeLocal<FunctionTemplate> ();
 
     auto tpl = maybeTpl.ToLocalChecked();
     auto *persistent = new Persistent<FunctionTemplate>(Isolate::GetCurrent(), tpl);
-    persistent->SetWeak (
-            g_base_info_ref (gi_info),
-            GObjectClassDestroyed,
-            WeakCallbackType::kParameter);
 
+    GType *gtypePtr = g_new(GType, 1);
+    persistent->SetWeak(gtypePtr, GObjectClassDestroyed,
+                        WeakCallbackType::kParameter);
     g_type_set_qdata(gtype, GNodeJS::template_quark(), persistent);
     return MaybeLocal<FunctionTemplate> (tpl);
 }
 
-static MaybeLocal<FunctionTemplate> GetClassTemplateFromGI(GIBaseInfo *info) {
+MaybeLocal<Function> MakeClass(GIBaseInfo *info) {
     GType gtype = g_registered_type_info_get_g_type ((GIRegisteredTypeInfo *) info);
 
-    if (gtype == G_TYPE_NONE) {
+    if (gtype == G_TYPE_NONE || gtype == G_TYPE_INVALID) {
         const char *error = g_module_error();
         Throw::GTypeNotFound(info, error);
-        return MaybeLocal<FunctionTemplate>();
+        return MaybeLocal<Function>();
     }
-
-    auto tpl = GetClassTemplate(info, gtype);
-
-    if (tpl.IsEmpty())
-        return MaybeLocal<FunctionTemplate> ();
-
-    return MaybeLocal<FunctionTemplate> (tpl);
-}
-
-MaybeLocal<Function> MakeClass(GIBaseInfo *info) {
-    auto tpl = GetClassTemplateFromGI (info);
+    auto tpl = GetClassTemplate(gtype);
```

Comment:
Minor formatting issue here, I'd add newlines like illustrated to make it less cramed together, it's more readable.

![Screenshot from 2020-05-26 22-28-58](https://user-images.githubusercontent.com/1423607/82972146-5baffd00-9fa2-11ea-95d3-383bd4d3316e.png)